File: .temp/flutter_app/lib/domain/usecase/get_expenses_usecase.dart
import '../models/expense.dart';
import '../repository/expense_repository.dart';

class GetExpensesUseCase {
  final ExpenseRepository _repository;

  GetExpensesUseCase(this._repository);

  Stream<List<Expense>> execute() {
    return _repository.getAllExpenses();
  }
}

File: shared/src/commonMain/kotlin/com/sans/finance/domain/usecase/GetExpensesUseCase.kt
package com.sans.finance.domain.usecase

import com.sans.finance.domain.model.Expense
import com.sans.finance.domain.repository.ExpenseRepository
import kotlinx.coroutines.flow.Flow


class GetExpensesUseCase constructor(
    private val repository: ExpenseRepository
) {
    operator fun invoke(): Flow<List<Expense>> {
        return repository.getAllExpenses()
    }
}
